### PR TITLE
Migrate to Bounded Vec and Derive MaxEncodedLen

### LINKED ISSUE #1034: Migrate collator-selection Module to BoundedVec
#917

### PR DESCRIPTION
**Pull request by brettkolodny**
Closes #1034

### CODE REVIEW

**Comment by brettkolodny** [comment]
There are already two types for `MaxCandidates` and `MaxInvulnerables` that are tied to the two storage items that need to be migrated to `BoundedVec` but those traits are used for benchmarking only.

Should these be left as `Vec` or should two new types be implemented that serve as maximum amount that should ever be possible?

**Comment by xlc** [comment]
We have make them strictly enforced

File: modules/collator-selection/src/lib.rs [hunk 1]
```diff
@@ -249,23 +243,10 @@ pub mod pallet {
 		pub fn set_invulnerables(origin: OriginFor<T>, new: Vec<T::AccountId>) -> DispatchResultWithPostInfo {
 			T::UpdateOrigin::ensure_origin(origin)?;
 			// we trust origin calls, this is just a for more accurate benchmarking
```
Comment (xlc) [inline modules/collator-selection/src/lib.rs:245]: this comment can go

**Review by xlc — CHANGES_REQUESTED** [review]
I think we still want desired_candidates as a soft limit can be increased without runtime upgrade.

**Review by xlc — APPROVED** [review]

**Pull request merged**